source/math/FloatEuclidTransform.ooc

```diff
@@ -97,12 +100,32 @@ FloatEuclidTransform: cover {
 	operator * (value: Float) -> This {
 		This new(this translationX * value, this translationY * value, this rotationX * value, this rotationY * value, this rotationZ * value, (this scaling - 1) * value + 1)
 	}
+	toImageCoordinates: func(zDistance: Float) -> This {
+		correction := Quaternion getTransform(this rotation, zDistance) * FloatPoint2D new(0.0f, 0.0f)
+		This new(
+			this translationX + correction x,
+			this translationY + correction y,
+			this rotationX,
+			this rotationY,
+			this rotationZ,
+			this scaling)
+	}
+	toCameraCoordinates: func(zDistance: Float) -> This {
+		correction := Quaternion getTransform(-this rotation, zDistance) * FloatPoint2D new(0.0f, 0.0f)
+		This new(
+			this translationX + correction x,
+			this translationY + correction y,
+			this rotationX,
+			this rotationY,
+			this rotationZ,
+			this scaling)
+	}
 	toString: func -> String {
-		" tx: " + "%8f" format(this translationX) +
-		" ty: " + "%8f" format(this translationY) +
-		" Scale: " + "%8f" format(this scaling) +
-		" Rx: " + "%8f" format(this rotationX) +
-		" Ry: " + "%8f" format(this rotationY) +
-		" Rz: " + "%8f" format(this rotationZ)
+		" tx: " >> "%8f" format(this translationX) >>
+		" ty: " >> "%8f" format(this translationY) >>
+		" Scale: " >> "%8f" format(this scaling) >>
+		" Rx: " >> "%8f" format(this rotationX) >>
+		" Ry: " >> "%8f" format(this rotationY) >>
+		" Rz: " >> "%8f" format(this rotationZ)
```

Comment:
Wow, I'd entirely forgotten about this: You can use `formatFloat` instead of `format`. Faster, no leaks, easy to use.